plugins/catalog-backend/src/service/DefaultEntitiesCatalog.ts

```diff
@@ -669,11 +658,8 @@ export class DefaultEntitiesCatalog implements EntitiesCatalog {
   }
 
   async entityAncestry(rootRef: string): Promise<EntityAncestryResponse> {
-    const [rootRow] = await this.database<DbRefreshStateRow>('refresh_state')
-      .leftJoin<DbFinalEntitiesRow>('final_entities', {
-        'refresh_state.entity_id': 'final_entities.entity_id',
-      })
-      .where('refresh_state.entity_ref', '=', rootRef)
+    const [rootRow] = await this.database<DbFinalEntitiesRow>('final_entities')
```

Comment:
Should we really change this function at all? Now you're changing its semantics to deal with already-stitched things, which makes it miss for example entities that have processing errors. The bodies will also be the stitched ones, which wasn't the case before.

Essentially I think if we want to make any changes to this function then maybe it should be in a separate PR and probably as a breaking change even (shunting it to a 2.0 worst case).

Reply:
I'm not sure I understand what you mean, the code always stitched with `final_entities`, it's just a reverse lookup with the `refresh_state` table from `entity_ref` to the actual `final_entities` row.

It's always used stitched data as far as I can tell?

Reply:
Ah I missed the join and only looked here locally. OK!